gridscale/common.go

```diff
@@ -0,0 +1,104 @@
+package gridscale
+
+import (
+	"fmt"
+	"github.com/gridscale/gsclient-go"
+	"time"
+)
+
+type gsService string
+
+const delayFetchingStatus = 500
+
+const (
+	provivisoningStatus = "in-provisioning"
+	activeStatus        = "active"
+)
+
+const (
+	loadbalancerService gsService = "loadbalancer"
+	ipService           gsService = "IP"
+	networkService      gsService = "network"
+	serverService       gsService = "server"
+	sshKeyService       gsService = "sshkey"
+	storageService      gsService = "storage"
+)
+
+type objectGetter func(id string) error
+
+//convSOStrings converts slice of interfaces to slice of strings
+func convSOStrings(interfaceList []interface{}) []string {
+	var labels []string
+	for _, labelInterface := range interfaceList {
+		labels = append(labels, labelInterface.(string))
+	}
+	return labels
+}
+
+func pauseWhenProvisoning(client *gsclient.Client, service gsService, id string) error {
```

Comment:
I think we should use this retry logic. https://www.terraform.io/docs/extend/resources/retries-and-customizable-timeouts.html#retry

Reply:
This case is different. The pause will only take place when the status of the object is "in-provisioning".